pkg/controllers/user/logging/utils/wrap.go

```diff
@@ -159,17 +160,52 @@ func getWrapConfig(es *v3.ElasticsearchConfig, sp *v3.SplunkConfig, sl *v3.Syslo
 			Host: host,
 			Port: port,
 		}
+		currentTarget = loggingconfig.Syslog
 	}
 
-	if kf != nil && len(kf.BrokerEndpoints) != 0 {
-		currentTarget = loggingconfig.Kafka
-		wkf = WrapKafka{
-			Brokers: strings.Join(kf.BrokerEndpoints, ","),
+	if kf != nil {
+		if len(kf.BrokerEndpoints) == 0 && kf.ZookeeperEndpoint == "" {
+			err = errors.New("one of the kafka endpoint must be set")
+			return
 		}
+		if len(kf.BrokerEndpoints) != 0 {
+			var bs []string
+			var h string
+			for _, v := range kf.BrokerEndpoints {
+				h, _, err = parseEndpoint(v)
+				if err != nil {
+					return
+				}
+				bs = append(bs, h)
+			}
+			wkf = WrapKafka{
+				Brokers: strings.Join(bs, ","),
+			}
+		} else {
+			if kf.ZookeeperEndpoint != "" {
+				if _, _, err = parseEndpoint(kf.ZookeeperEndpoint); err != nil {
+					return
```

Comment:
same here - is it worth logging the error?

Reply:
it will return the message to UI, when the validate is failed, may be don't need to log the error.